pkg/cloudprovider/aws/amifamily/userData.go

```diff
@@ -0,0 +1,56 @@
+/*
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package amifamily
+
+import (
+	"context"
+	"fmt"
+
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+
+	"github.com/aws/karpenter/pkg/cloudprovider/aws/apis/v1alpha1"
+)
+
+type UserDataProvider struct {
+	clientSet *kubernetes.Clientset
+}
+
+// New constructs a new launch template Resolver
+func NewUserDataProvider(clientSet *kubernetes.Clientset) *UserDataProvider {
+	return &UserDataProvider{
+		clientSet: clientSet,
+	}
+}
+
+// Get returns a set of AMIIDs and corresponding instance types. AMI may vary due to architecture, accelerator, etc
+func (u *UserDataProvider) Get(ctx context.Context, provider *v1alpha1.AWS) (string, error) {
```

Comment:
this probably works for now, I think we'll eventually want a controller to watch the config maps to know when they go out-of-sync but that can be part of the out-of-spec node updates scope. 

Actually, maybe we only allow immutable config maps to make this easier to reason about and easier for Karpenter to handle: https://kubernetes.io/docs/concepts/configuration/configmap/#configmap-immutable

Reply:
> I think we'll eventually want a controller to watch the config maps to know when they go out-of-sync

Why do we need a controller to detect that? The version we're caching has the entire UserData content, not just the configmap name + namespace so we should use the latest content during every launch.

I think another controller is probably worth it just to introduce some caching and preventing that getConfigMap API call for latency reasons, but I figured that optimization can come later. 

I also just realized that the comment on `Get` is outdated so I'll fix that too :P

Reply:
Added a dedicated informer for ConfigMaps. Thanks for the pointers!